### DISCUSSION TITLE
Library of alternate data services

### DISCUSSION THREAD

**Discussion by windbeneathyourwings** [ngrx/platform#3170] [Q&A]
Currently data supports basic REST out the box. I was wondering if there is any interest or library already available of alternate data services pre-built. For example, I'm currently saving documents to aws s3 as json. This is done through a lambda running on aws. I have plans to convert that to a completely serverless model. Instead of using the lambda replace that with client-side signed url/cookie to put the object on s3. No longer needing the lambda and greatly reducing the complexity of my project since the back-end api is no longer necessary for master entities stored as json on s3. To achieve this goal in the most elegant manor possible I would like to replace the standard REST data service with one that persists to s3. I've done a bit of research and came up empty. It's difficult to believe no libraries of alternate data services exists. In the age of serverless development it would be nice to have a repository/project of well groomed alternate data services. Not just for aws storages but others as well like elastic. Plenty of modern storages allow directly persisting entities without needing a middle, server REST api layer.

**Comment by windbeneathyourwings** [comment]
Another example of something that would be useful is saving to local, client data stores like RxDB.

**Comment by timdeschryver** [comment]
This isn't on our roadmap.
Feel free to ask this question on our [discord server](https://discord.com/invite/ngrx), perhaps that someone knows a library, or already has some sample code to do this.
For now, I'm converting this issue to a discussion.
